### PR TITLE
Fix: bolt-button text and icon alignment

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

http://vjira2:8080/browse/BDS-791

## Summary

Improves the vertical alignment of text and icon inside `bolt-button`.

## Details

1. CSS are rewritten to handle vertical alignment of the text and icon. 
2. Icon-only option now has consistent spacing and size (perfect square/circle), and the icon inside will scale based on button size.
3. Documentation is rewritten to demonstrate each prop properly and clearly.
4. Folder structure and internal variables have been converted to current standards.
5. There should be no breaking changes, future tasks have been created in JIRA to cover further refactoring and cleanup that would break existing buttons.

## How to test

Run the branch locally and check to see if all the demos are working as intended.

### CODE REVIEW

**Comment by sghoweri** [comment]
@mikemai2awesome can you try running `npm run test` on this branch locally? 

If the button component tests end up failing, you can try running `npx jest packages/components/bolt-button/__tests__/button.js -u` to update the Jest snapshots, then retest to confirm via `npm run test`. If that all checks out (and the snapshot HTML is correct), then you can just commit the updated Jest snapshot code and we should be all set.

File: docs-site/src/pages/pattern-lab/_patterns/02-components/button/999-button-with-web-component.twig [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  url="https://pega.com"
+  size="large"
+  color="secondary"
+  width="auto"
+  border-radius="full"
+  align="center"
+  transform="uppercase"
+  tag="a"
+  icon-only
+>
```
Comment (sghoweri) [inline docs-site/src/pages/pattern-lab/_patterns/02-components/button/999-button-with-web-component.twig]: @mikemai2awesome is it worth mentioning that the `width`, `align`, `transform`, and `tag` props for this example are all not actually needed to get this button example working? Rather, these are just getting added for demonstration purposes.

**Comment by mikemai2awesome** [comment]
@sghoweri 

The consistent focus style is implemented, it's the same as input focus, therefore the browser outline is not needed. You can see it in the focus demo.

File: docs-site/src/pages/pattern-lab/_patterns/02-components/button/999-button-with-web-component.twig (hunk 1 above)
Comment (mikemai2awesome) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/button/999-button-with-web-component.twig]: Added some text to clarify the usage of props.

**Comment by sghoweri** [comment]
> 

> @sghoweri
> 
> The consistent focus style is implemented, it's the same as input focus, therefore the browser outline is not needed. You can see it in the focus demo.

Thanks. I just checked out your latest updates - much better! 👍

**Pull request merged**